### ISSUE TITLE
[turbopack] The browser page freezes after changing the code

### ISSUE THREAD

**Issue by BigTomM** [vercel/turbo#5698]
### What version of Next.js are you using Turbopack in?

13.4.4

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Windows

### Describe the Bug

After referencing a common function in multiple pages, for example, I used a public utility function on 5 pages, which means importing the function of the "/utils" folder into the "/pages" folder, and then I modified the function's code, the page was stuck, Chrome displays that the CPU usage of this tab is 100%. But if I turn off the --turbo mode, there is no problem.

### Expected Behavior

The browser page should update the latest content returned by the function

### To Reproduce

1. import a common funtion in multiple pages;
2. then change the common funtion's code and save;
3. reproduce!

**Comment by padmaia** [comment]
@BigTomM can you please provide a reproduction repo 🙏

**Comment by timneutkens** [comment]
Going to close this issue as the reproduction is missing and there's been many changes to Turbopack over the past few months to improve reliability.

**Comment by monojack** [comment]
@msdrigg, I was having the same issue, where the UI would freeze entirely when changing/saving code with `next dev --turbo`, though I noticed it happened consistently when changing a specific module. So I thought that maybe circular dependency would be an issue. 

I used [madge](https://github.com/pahen/madge) to scan for circular dependencies and, sure enough, it found one. It was not a direct circular dependency, but more like module `A` would import `B`, `B` -> `C`, `C` -> `D`, `D` -> `E` and then `E` would import `A`. After fixing it, the UI no longer freezes.
